Structure_oM/SectionProperties/Reinforcement/BarRebarIntent.cs

```diff
@@ -0,0 +1,48 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Quantities.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+
+namespace BH.oM.Structure.SectionProperties.Reinforcement
+{
+    [Description("A collection of reinforcement for a Bar and minimum cover.")]
+    public class BarRebarIntent : BHoMObject, IRebarIntent
+    {
+        /***************************************************/
+        /**** Properties                                ****/
+        /***************************************************/
+
+        [Description("A collection of IBarReinforcement objects.")]
+        public virtual List<IBarReinforcement> BarReinforcement { get; set; }
```

Comment:
Usually the properties with a setter are initialised in the declaration. This is particularly helpful in case of collections, to avoid unexpected nulls. Not sure if this is just an observation or a solid recommendation, maybe @FraserGreenroyd could tell more?
```suggestion
        public virtual List<IBarReinforcement> BarReinforcement { get; set; } = new List<IBarReinforcement>();
```